subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/JavaToolchain.java

```diff
@@ -84,7 +85,7 @@ public JavadocTool getJavadocTool() {
         return toolFactory.create(JavadocTool.class, this);
     }
 
-    @Internal
+    @Input
```

Comment:
We are making this an Input, because the language version in the `JavaToolchainInput` can actually be null for the `SpecificInstallationToolchainSpec` and therefore is optional.

Reply:
You'll need to create an integration test which shows that a change here causes a task to be out of date.

Reply:
It seems it has been an input already, just in the interface. However, the implementation was missing the `@Override` annotation.